app/controllers/conversations_controller.rb

```diff
@@ -942,8 +942,14 @@ def render_error(attribute, message)
   end
 
   def infer_scope
-    filter_mode = (params[:filter_mode].respond_to?(:to_sym) && params[:filter_mode].to_sym) || :or
-    return render_error('filter_mode', 'invalid') if ![:or, :and].include?(filter_mode)
+    # John Stauffacher (john.stauffacher@gmail.com) prevents an unsafe cast to sym 
+    if params[:filter_mode].respond_to?(:to_sym)
+    	if ['or','and'].include? params[:filter_mode]
+    		filter_mode = (params[:filter_mode].respond_to?(:to_sym) && params[:filter_mode].to_sym) 
+    	else
+    		return render_error('filter_mode', 'invalid')
+	end
+    end
```

Comment:
some initial stylistic feedback:
- we prefer 2 spaces over tabs
- remove trailing whitespace